docs/key-concepts.md

```diff
@@ -0,0 +1,122 @@
+
+Observability and monitoring essentially mean keeping a close eye on how systems behave and perform using logs and metrics. These logs and metrics are generated by various components within the system, including applications, servers, databases, and networking devices. They serve as valuable sources of information for understanding system health, diagnosing issues, and optimizing performance.
+
+In this module, we’ll explore the ***key concepts of observability*** and understand how logs and metrics play a crucial role in providing insights into system behaviour and performance.
+
+## 1. Logs 📝
+
+Logs are simply the **records of events** or actions that occur within a system. Each log entry typically contains relevant information such as **timestamps**, **event descriptions**, **severity levels**, and **contextual data**. Logs serve as a detailed history of system activity, providing valuable insights into system behaviour, errors, and performance issues.
+
+These logs are essential for troubleshooting problems, auditing system activity, and understanding user interactions. In software development and operations, logs are generated by various components, including applications, servers, databases, and networking devices.
+
+This is what a sample log data in a system would look like.
+
+| Timestamp            | Event Description                       | Severity | Contextual Data                                    |
+| -------------------- | --------------------------------------- | -------- | -------------------------------------------------- |
+| 2024-02-11T08:35:00Z | Application server started successfully | INFO     | Server: app-01, Environment: Production            |
+| 2024-02-11T09:12:45Z | Disk usage exceeded threshold           | WARNING  | Server: db-03, Disk Usage: 85%, Threshold: 80%     |
+| 2024-02-11T10:20:30Z | Critical error in database connection   | ERROR    | Server: db-02, Error Code: 500, Connection Timeout |
+| 2024-02-11T11:05:15Z | User authentication failed              | ERROR    | Username: johndoe, IP: 192.168.1.100               |
+| 2024-02-11T12:30:00Z | Scheduled job completed successfully    | INFO     | Job: Backup, Duration: 00:45:00                    |
+
+### Log ingestion
+
+As the name suggests, it refers to the process of **collecting and importing log data** from various sources into a centralized system for analysis and storage. It involves capturing log events generated by applications, servers, and other components, parsing and enriching the log data, and forwarding it to a centralized logging system or storage backend. Log ingestion systems often support various protocols and formats for data collection, including syslog, JSON, and structured logging.
```

Comment:
```suggestion
As the name suggests, it refers to the process of **collecting and importing log data** from various sources into a centralized system for analysis and storage. It involves capturing log events generated by applications, servers, and other components, parsing and enriching the log data, and forwarding it to a centralized logging system or storage backend. Log ingestion systems often support various protocols and formats for data collection, including syslog, JSON, and structured logging.
```

This is slightly incorrect. Log ingestion doesn't involve capturing data. The capturing/generation of telemetry data is done by collectors. Ingestion is simply the process of sending the collected data to the centralized server.

Within Logs itself, let's have two more points. One about `Collectors` and another about `Retention` (period for how long the telemetry data is stored)

Reply:
Thanks for the insight, I'll research and make it more accurate

Yes, I'll add more info on Collectors and Retention